handlers/auth.go

```diff
@@ -0,0 +1,112 @@
+package handlers
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"time"
+
+	"github.com/Daniorocket/cookit-backend/lib"
+	"github.com/Daniorocket/cookit-backend/models"
+	uuid "github.com/satori/go.uuid"
+	"golang.org/x/crypto/bcrypt"
+)
+
+func (d *Handler) Login(w http.ResponseWriter, r *http.Request) {
+	login := &models.Login{}
+	err := json.NewDecoder(r.Body).Decode(login)
+	if err != nil {
+		log.Println("Error:", err)
+		if err := CreateApiResponse(w, nil, http.StatusBadRequest, "failed", "Failed to decode json"); err != nil {
+			log.Println("Error create Api response:", err)
+		}
+		return
+	}
+	passDB, err := models.GetPasswordByUsernameOrEmail(d.Client, d.DatabaseName, login.Username)
+	if err != nil {
+		log.Println("Error:", err)
+		if err := CreateApiResponse(w, nil, http.StatusBadRequest, "failed", "Invalid username or password"); err != nil {
+			log.Println("Error create Api response:", err)
+		} //User doesn't exists
+		return
+	}
+	if err := bcrypt.CompareHashAndPassword([]byte(passDB), []byte(login.Password)); err != nil {
+		log.Println("Error:", err)
+		if err := CreateApiResponse(w, nil, http.StatusBadRequest, "failed", "Invalid username or password"); err != nil {
+			log.Println("Error create Api response:", err)
+		}
+		return
+	}
+	expirationTime := time.Now().Add(15 * time.Minute)
+	tokenString, err := lib.CreateJWT(login.Username, expirationTime)
+	if err != nil {
+		CreateApiResponse(w, nil, http.StatusBadRequest, "failed", "Failed to create JWT")
+		return
+	}
+	CreateApiResponse(w, lib.Token{
+		Token:          tokenString,
+		ExpirationTime: expirationTime.UTC().String(),
+	}, http.StatusOK, "success", "none")
+}
+func (d *Handler) Register(w http.ResponseWriter, r *http.Request) {
+	cred := models.Credentials{}
+	if err := json.NewDecoder(r.Body).Decode(&cred); err != nil {
+		log.Println("Error:", err)
+		if err := CreateApiResponse(w, nil, http.StatusBadRequest, "failed", "Failed to decode json"); err != nil {
+			log.Println("Error create Api response:", err)
+		}
+		return
+	}
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(cred.Password), 8)
```

Comment:
Czym jest magiczna `8`? 
Te akcje `CompareHashAndPassword`, `GenerateFromPassword` bym opakował w osobny pakiet (często nazwa to po prostu `hasher`) i wtedy taka magiczna 8 nie będzie w handlerze i łatwiej będzie Ci zmockować `CompareHashAndPassword` do testów jak zrobisz interfejs na to :)

Reply:
To z tą magiczną ósemką, to jest liczba oznaczona jako cost. Pamiętam, że pisząc ten fragment kodu wspierałem się jakimiś rozwiązaniami, odnośnie autoryzacji. Przeniosę w takim razie te rzeczy do osobnego package oraz usunę magiczną zmienną.